tests/test_smartbridge.py

```diff
@@ -269,6 +276,102 @@ async def _accept_connection(self, leap, wait):
+            return
+
+        for station in result.Body.get("ControlStations", []):
+            for device in station.get("AssociatedGangedDevices", []):
+                device_type = device["Device"]["DeviceType"]
+                if device_type not in _LEAP_DEVICE_TYPES.get("sensor"):
+                    continue
+
+                device_id = re.sub(r".*/", "", device["Device"]["href"])
+                request, response = await wait(leap.requests.get())
+                assert request == Request(
+                    communique_type="ReadRequest",
+                    url=f"/device/{device_id}/buttongroup/expanded",
+                )
+                button_group_result = response_from_json_file(
+                    f"ra3-devicebg{device_id}.json"
+                )
+                response.set_result(button_group_result)
+                leap.requests.task_done()
+
+                request, response = await wait(leap.requests.get())
+                assert request == Request(
+                    communique_type="ReadRequest", url=f"/device/{device_id}"
+                )
+                response.set_result(
+                    response_from_json_file(f"ra3-device{device_id}.json")
+                )
+                leap.requests.task_done()
+
+                # collect every button for upcoming subscribe tests
+                self.ra3_button_list.extend(
+                    [
+                        id_from_href(button["href"])
+                        for group in button_group_result.Body["ButtonGroupsExpanded"]
+                        for button in group["Buttons"]
+                    ]
+                )
+
+    async def _accept_connection_ra3(self, leap, wait):
+        """Accept a connection from SmartBridge (implementation)."""
+        # Read request on /areas
+        ra3_area_list_result = response_from_json_file("ra3-areas.json")
+        request, response = await wait(leap.requests.get())
+        assert request == Request(communique_type="ReadRequest", url="/area")
+        response.set_result(ra3_area_list_result)
+        leap.requests.task_done()
+
+        # Read request on /device
+        request, response = await wait(leap.requests.get())
+        assert request == Request(communique_type="ReadRequest", url="/device")
+        response.set_result(response_from_json_file("ra3-devices.json"))
+        leap.requests.task_done()
+
+        # Read request on each area's control stations & zones
+        for area_id in (
+            re.sub(r".*/", "", area["href"])
+            for area in ra3_area_list_result.Body.get("Areas", [])
+            if area["IsLeaf"]
+        ):
+            request, response = await wait(leap.requests.get())
+            assert request == Request(
+                communique_type="ReadRequest",
+                url=f"/area/{area_id}/associatedcontrolstation",
+            )
+            station_result = response_from_json_file(f"ra3-area{area_id}cs.json")
```

Comment:
Would it make more sense to use folders? `ra3/area/{area_id}/station.json`?

Reply:
I've revised the responses into a directory structure like you've suggested